includes/wc-template-functions.php

```diff
@@ -512,6 +512,124 @@ function wc_product_post_class( $classes, $class = '', $post_id = '' ) {
+			$class = preg_split( '#\s+#', $class );
+		}
+		$classes = array_map( 'esc_attr', $class );
+	} else {
+		// Ensure that we always coerce class to being an array.
+		$class = array();
+	}
+
+	if ( ! $post ) {
+		return $classes;
+	}
+
+	$classes[] = 'post-' . $post->ID;
+	if ( ! is_admin() ) {
+		$classes[] = $post->post_type;
+	}
+	$classes[] = 'type-' . $post->post_type;
+	$classes[] = 'status-' . $post->post_status;
+
+	// Post format.
+	if ( post_type_supports( $post->post_type, 'post-formats' ) ) {
+		$post_format = get_post_format( $post->ID );
+
+		if ( $post_format && ! is_wp_error( $post_format ) ) {
+			$classes[] = 'format-' . sanitize_html_class( $post_format );
+		} else {
+			$classes[] = 'format-standard';
+		}
+	}
+
+	// Post requires password.
+	$post_password_required = post_password_required( $post->ID );
+	if ( $post_password_required ) {
+		$classes[] = 'post-password-required';
+	} elseif ( ! empty( $post->post_password ) ) {
+		$classes[] = 'post-password-protected';
+	}
+
+	// Post thumbnails.
+	if ( current_theme_supports( 'post-thumbnails' ) && has_post_thumbnail( $post->ID ) && ! is_attachment( $post ) && ! $post_password_required ) {
+		$classes[] = 'has-post-thumbnail';
+	}
+
+	// Sticky for Sticky Posts.
+	if ( is_sticky( $post->ID ) ) {
+		if ( is_home() && ! is_paged() ) {
+			$classes[] = 'sticky';
+		} elseif ( is_admin() ) {
+			$classes[] = 'status-sticky';
+		}
+	}
+
+	// Hentry for hAtom compliance.
+	$classes[] = 'hentry';
+
+	// All public taxonomies.
+	if ( apply_filters( 'woocommerce_get_product_class_include_taxonomies', false ) ) {
+		$taxonomies = get_taxonomies( array( 'public' => true ) );
+		foreach ( (array) $taxonomies as $taxonomy ) {
+			if ( is_object_in_taxonomy( $post->post_type, $taxonomy ) ) {
+				foreach ( (array) get_the_terms( $post->ID, $taxonomy ) as $term ) {
+					if ( empty( $term->slug ) ) {
+						continue;
+					}
+
+					$term_class = sanitize_html_class( $term->slug, $term->term_id );
+					if ( is_numeric( $term_class ) || ! trim( $term_class, '-' ) ) {
+						$term_class = $term->term_id;
+					}
+
+					// 'post_tag' uses the 'tag' prefix for backward compatibility.
+					if ( 'post_tag' === $taxonomy ) {
+						$classes[] = 'tag-' . $term_class;
+					} else {
+						$classes[] = sanitize_html_class( $taxonomy . '-' . $term_class, $taxonomy . '-' . $term->term_id );
+					}
+				}
+			}
+		}
+	}
+
+	$classes = array_map( 'esc_attr', $classes );
+	$classes = apply_filters( 'post_class', $classes, $class, $post->ID );
+
+	return array_unique( $classes );
+}
+
+/**
+ * Display the classes for the product div.
+ *
+ * @since 3.4.0
+ * @param string|array           $class      One or more classes to add to the class list.
+ * @param int|WP_Post|WC_Product $product_id Product ID or product object.
+ */
+function wc_product_class( $class = '', $product_id = null ) {
+	echo 'class="' . join( ' ', wc_get_product_class( $class, $product_id ) ) . '"'; // WPCS: XSS ok.
```

Comment:
Not ok :) Classes coming in from `post_class` are not escaped. I'd prefer you escaped here rather than in `wc_get_product_class`

Reply:
It's escaped inside `wc_get_product_class()` with `$classes = array_map( 'esc_attr', $classes );`.
It's just the same like `post_class()` of WordPress.